Add support for optional kafka SASL (#24798)
prestodb/presto: presto-docs/src/main/sphinx/connector/kafka.rst

``etc/catalog/kafka.properties`` with the following contents,
replacing the properties as appropriate:

In some cases, such as when using specialized authentication methods, it is necessary to specify

> additional Kafka client properties in order to access your Kafka cluster. To do so,

Author: Done.

> add the `kafka.config.resources` property to reference your Kafka config files. Note that configs

Author: Done

> can be overwritten if defined explicitly in `kafka.properties`:

Author: Done

> can be overwritten if defined explicitly in `kafka.properties`:

Author: `Does this mean "Configuration settings in the files in kafka.config.resources can be overridden if specified explicitly in kafka.properties." ?`

Yes, I have updated the sentence to make it a bit more meaningful and clear, as shown above, and provided an example.

``etc/catalog/kafka.properties`` with the following contents,
replacing the properties as appropriate:

In some cases, such as when using specialized authentication methods, it is necessary to specify
additional Kafka client properties to access your Kafka cluster. To do so,

> add the ``kafka.config.resources`` property to reference your Kafka config files.

Reviewer: ```suggestion
add the ``kafka.config.resources`` property to reference your Kafka configuration files.
```

Configuration settings in the files in ``kafka.config.resources`` can be overridden if

> specified explicitly in ``kafka.properties`` (e.g., security.protocol=SSL):

Reviewer: ```suggestion
specified explicitly in ``kafka.properties`` (for example, ``security.protocol=SSL``):
```
Avoid Latin abbreviations and phrases. See the [Gitlab Documentation Style Guide Recommended Word List entry for e.g.](https://docs.gitlab.com/development/documentation/styleguide/word_list/#eg) for suggestions of alternatives.

This property is optional; the default is ``true``.

``kafka.security-protocol``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Protocol used to communicate with brokers.

> Valid values are: SASL_PLAINTEXT, SASL_SSL.

Author: Done.

This property is optional; default is ``SASL_PLAINTEXT``.

``kafka.sasl.mechanism``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> This specifies the authentication mechanism used by SASL.

Author: Done.

Valid values are: SCRAM, PLAIN ...

This property is optional.

``kafka.sasl.jaas.config``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> This config used for SASL authentication, which is a method for securing communication with brokers.

Author: Done

This property is optional; the default is ``true``.

``kafka.security-protocol``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Protocol used to communicate with brokers.
Valid values are: ``SASL_PLAINTEXT``, ``SASL_SSL``.

This property is optional; default is ``SASL_PLAINTEXT``.

``kafka.sasl.mechanism``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

The authentication mechanism used by SASL.
Valid values are: ``PLAIN``, ``SCRAM-SHA-256``, ``SCRAM-SHA-512``.

This property is optional.

``kafka.sasl.jaas.config``
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> This configuration used for SASL authentication, which is a method for securing communication with brokers.

Reviewer: ```suggestion
The configuration used for SASL authentication, which is a method for securing communication with brokers.
```

> This property is optional, but required when ``kafka.sasl.mechanism`` is given.

Reviewer: ```suggestion
This property is optional, but is required when ``kafka.sasl.mechanism`` is given.
```

``kafka.truststore.path``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Path of the truststore file used for connection to Kafka cluster.

Reviewer: ```suggestion
Path of the truststore file used for connecting to the Kafka cluster.
```
